Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/aspect/BaseEventAdviceDelegate.java

```diff
@@ -243,24 +238,36 @@ public void createNewEvent() {
     private AssertionType getAssertion(Object[] args) {
         AssertionType assertion = null;
         for (Object obj : args) {
-            if (obj instanceof AssertionType) {
-                assertion = (AssertionType) obj;
-            } else if (obj instanceof RespondingGatewayPRPAIN201305UV02RequestType) {
-                assertion = ((RespondingGatewayPRPAIN201305UV02RequestType) obj).getAssertion();
-            } else if (obj instanceof RespondingGatewayCrossGatewayQueryRequestType) {
-                assertion = ((RespondingGatewayCrossGatewayQueryRequestType) obj).getAssertion();
-            } else if (obj instanceof RespondingGatewayCrossGatewayRetrieveRequestType) {
-                assertion = ((RespondingGatewayCrossGatewayRetrieveRequestType) obj).getAssertion();
-            } else if (obj instanceof RespondingGatewayProvideAndRegisterDocumentSetRequestType) {
-                assertion = ((RespondingGatewayProvideAndRegisterDocumentSetRequestType) obj).getAssertion();
-            } else if (obj instanceof RespondingGatewayProvideAndRegisterDocumentSetResponseRequestType) {
-                assertion = ((RespondingGatewayProvideAndRegisterDocumentSetResponseRequestType) obj).getAssertion();
-            } else if (obj instanceof RespondingGatewaySendAlertMessageType) {
-                assertion = ((RespondingGatewaySendAlertMessageType) obj).getAssertion();
-            } else if (obj instanceof RespondingGatewayPRPAIN201306UV02RequestType) {
-                assertion = ((RespondingGatewayPRPAIN201306UV02RequestType) obj).getAssertion();
+            assertion = getAssertion(obj);
+            if (null != assertion) {
+                break;
             }
         }
         return assertion;
     }
+
+    public AssertionType getAssertion(Object obj) {
+        try {
+            if (obj != null) {
+                if (obj instanceof AssertionType) {
+                    return (AssertionType) obj;
+                } else {
+                    return getAssertionType(obj.getClass().getDeclaredMethods(), obj);
+                }
+            }
+        } catch (IllegalAccessException | InvocationTargetException ex) {
+            LOG.info("Unable to extract AssertionType from Object", ex);
```

Comment:
Can we log the object as well? At the very least the object's class. I would also consider this to be an "Error" level, not Info.